### PR TITLE
payday - changing tax and subtotal calculation

### PR DESCRIPTION
**Pull request by katiaAlexandrov**

### CODE REVIEW

File: lib/payday/invoiceable.rb [hunk 1]
```diff
@@ -20,18 +20,22 @@ def bill_to
 
   # Calculates the subtotal of this invoice by adding up all of the line items
   def subtotal
-    line_items.reduce(BigDecimal.new("0")) { |result, item| result += item.amount }
+    line_items.reduce(BigDecimal.new("0")) { |result, item| result += item.amount_no_tax }
```
Comment (alonhartal) [inline lib/payday/invoiceable.rb:23]: @katiaAlexandrov going back to this branch - where is subtotal_with_items_tax and  items_tax going to be used? this should be changed on the backend as well?

Comment (katiaAlexandrov) [reply lib/payday/invoiceable.rb:23]: @alonhartal it's used internally, before the change it was only subtotal for line, so I've splitted It to amount_no_tax + item_tax + amount.
Total tax is sum of all lines item_tax
Subtotal is sum of all lines amount_no_tax.

File: lib/payday/invoiceable.rb [hunk 2]
```diff
@@ -20,18 +20,22 @@ def bill_to
...
   # The tax for this invoice, as a BigDecimal
   def tax
-    if defined?(tax_rate)
-      calculated = subtotal * tax_rate
-      return 0 if calculated < 0
-      calculated
-    else
-      0
-    end
+      calculated = subtotal_with_items_tax * tax_rate
```
Comment (katiaAlexandrov) [inline lib/payday/invoiceable.rb]: @alonhartal I've reviewed this one more time, it's very not clear.
In general, there are 2 possible taxes in payday - per line and general tax rate. In any case you should not use both, we use tax per line only. I think I will remove all this logic from tax that related to general tax_rate, it's confusing and no purpose for this. What do u think?

Comment (alonhartal) [reply lib/payday/invoiceable.rb]: makes sense for me @katiaAlexandrov

File: lib/payday/invoiceable.rb (hunk 1 above)
Comment (alonhartal) [reply lib/payday/invoiceable.rb:23]: @katiaAlexandrov can you share a PDF example with the taxes?

**Pull request merged**